src/react-components/room/useResizeViewport.js

```diff
@@ -38,84 +38,70 @@ export function useResizeViewport(viewportRef, store, scene) {
     height: getMaxResolutionHeight(store)
   });
 
-  useEffect(
-    () => {
-      function onStoreChanged() {
-        setMaxResolution({
-          width: getMaxResolutionWidth(store),
-          height: getMaxResolutionHeight(store)
-        });
-      }
+  useEffect(() => {
+    function onStoreChanged() {
+      setMaxResolution({
+        width: getMaxResolutionWidth(store),
+        height: getMaxResolutionHeight(store)
+      });
+    }
+
+    onStoreChanged();
 
-      onStoreChanged();
+    store.addEventListener("statechanged", onStoreChanged);
 
-      store.addEventListener("statechanged", onStoreChanged);
+    return () => {
+      store.removeEventListener("statechanged", onStoreChanged);
+    };
+  }, [store]);
 
-      return () => {
-        store.removeEventListener("statechanged", onStoreChanged);
-      };
-    },
-    [store]
-  );
+  useEffect(() => {
+    const observer = new ResizeObserver(entries => {
+      const isPresenting = scene.renderer.xr.isPresenting;
+      const isVRPresenting = scene.renderer.xr.enabled && isPresenting;
 
-  useEffect(
-    () => {
-      const observer = new ResizeObserver(entries => {
-        const isPresenting = scene.renderer.xr.isPresenting;
-        const isVRPresenting = scene.renderer.xr.enabled && isPresenting;
+      // Do not update renderer, if a camera or a canvas have not been injected.
+      // In VR mode, three handles canvas resize based on the dimensions returned by
+      // the getEyeParameters function of the WebVR API. These dimensions are independent of
```

Comment:
Does this mean Three.js still handles WebVR API? I thought it has dropped WebVR API support...

Reply:
I think this comment is just out of date. It is still true though that WebXRManager handles the size of its framebuffers internally.